### PR TITLE
pillow - Buffer Overflow

### PR DESCRIPTION
**Pull request by spolti**
chore: Fixes [CVE-2024-28219](https://www.cve.org/CVERecord?id=CVE-2024-28219)

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

**Review by lizzzcai — APPROVED** [review]
/lgtm

File: hack/generate-licenses.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ mv license.txt third_party/library/license_go.txt
 
 ## Generate a Python License
 # See https://github.com/kubeflow/testing/blob/master/py/kubeflow/testing/python-license-tools/README.md
-pipenv install -e python/alibiexplainer python/kfserving python/sklearnserver python/xgbserver
+pipenv install -e python/kfserving python/sklearnserver python/xgbserver
```
Comment (yuzisun) [inline hack/generate-licenses.sh:15]:
```suggestion
pipenv install -e python/kserve python/sklearnserver python/xgbserver
```

Comment (spolti) [reply hack/generate-licenses.sh:15]: nice catch :)

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Comment by spolti** [comment]
hmm, seems that DCO is not happy, @yuzisun should I re-sign the commit or can we skip the DCO?

**Comment by yuzisun** [comment]
> hmm, seems that DCO is not happy, @yuzisun should I re-sign the commit or can we skip the DCO?

you can re-sign the commit, also need to resolve the conflicts

**Comment by yuzisun** [comment]
@spolti Can you help resolve the conflicts ? thanks

**Comment by spolti** [comment]
> @spolti Can you help resolve the conflicts ? thanks

Done, sry for the delay.

File: python/kserve/poetry.lock [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# This file is automatically @generated by Poetry 1.8.2 and should not be changed by hand.
+# This file is automatically @generated by Poetry 1.7.1 and should not be changed by hand.
```
Comment (yuzisun) [inline python/kserve/poetry.lock]: @spolti Let's use poetry 1.8?

Comment (spolti) [reply python/kserve/poetry.lock]: sure, let me update my end.

Done. Thanks for the heads up.

**Comment by spolti** [comment]
The e2e test is failing due:

```
2024-06-25 18:09:32.426 1 kserve INFO [storage.py:download():66] Copying contents of /mnt/models to local
2024-06-25 18:09:32.426 1 kserve INFO [storage.py:download():110] Successfully copied /mnt/models to None
2024-06-25 18:09:32.426 1 kserve INFO [storage.py:download():111] Model downloaded in 0.0005138100000294799 seconds.
2024-06-25 18:09:32.426 1 kserve ERROR [__main__.py:<module>():39] fail to locate model file for model isvc-sklearn-mms-v1 under dir /mnt/models, trying loading from model repository.
Traceback (most recent call last):
  File "/sklearnserver/sklearnserver/__main__.py", line 35, in <module>
    model.load()
  File "/sklearnserver/sklearnserver/model.py", line 46, in load
    raise ModelMissingError(model_path)
kserve.errors.ModelMissingError: <exception str() failed>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/sklearnserver/sklearnserver/__main__.py", line 44, in <module>
    kserve.ModelServer(
  File "/kserve/kserve/model_server.py", line 241, in start
    raise NoModelReady(models)
kserve.errors.NoModelReady: Models with name isvc-sklearn-mms-v1 are not ready.

```

not sure what is causing it. @sivanantha321 @yuzisun @terrytangyuan ideas?

**Comment by sivanantha321** [comment]
> The e2e test is failing due:
> 
> ```
> 2024-06-25 18:09:32.426 1 kserve INFO [storage.py:download():66] Copying contents of /mnt/models to local
> 2024-06-25 18:09:32.426 1 kserve INFO [storage.py:download():110] Successfully copied /mnt/models to None
> 2024-06-25 18:09:32.426 1 kserve INFO [storage.py:download():111] Model downloaded in 0.0005138100000294799 seconds.
> 2024-06-25 18:09:32.426 1 kserve ERROR [__main__.py:<module>():39] fail to locate model file for model isvc-sklearn-mms-v1 under dir /mnt/models, trying loading from model repository.
> Traceback (most recent call last):
>   File "/sklearnserver/sklearnserver/__main__.py", line 35, in <module>
>     model.load()
>   File "/sklearnserver/sklearnserver/model.py", line 46, in load
>     raise ModelMissingError(model_path)
> kserve.errors.ModelMissingError: <exception str() failed>
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
>     return _run_code(code, main_globals, None,
>   File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
>     exec(code, run_globals)
>   File "/sklearnserver/sklearnserver/__main__.py", line 44, in <module>
>     kserve.ModelServer(
>   File "/kserve/kserve/model_server.py", line 241, in start
>     raise NoModelReady(models)
> kserve.errors.NoModelReady: Models with name isvc-sklearn-mms-v1 are not ready.
> ```
> 
> not sure what is causing it. @sivanantha321 @yuzisun @terrytangyuan ideas?

Fix is raised #3758

**Comment by spolti** [comment]
@sivanantha321 cool, that was fast, many thanks.

**Pull request merged**